src/Database/Behaviors/Purgeable.php

```diff
@@ -0,0 +1,133 @@
+<?php namespace October\Rain\Database\Behaviors;
+
+use Exception;
+
+class Purgeable extends \October\Rain\Extension\ExtensionBase
+{
+    /**
+     * @var array List of attribute names which should not be saved to the database.
+     *
+     * protected $purgeable = [];
+     */
+
+    protected $parent;
+
+    public function __construct($parent)
+    {
+        $this->parent = $parent;
+        $this->bootPurgeable();
+    }
+
+    /**
+     * @var array List of original attribute values before they were purged.
+     */
+    protected $originalPurgeableValues = [];
+
+    /**
+     * Boot the purgeable trait for a model.
+     * @return void
+     */
+    public function bootPurgeable()
+    {
+        if (!$this->parent->propertyExists('purgeable'))
+        {
+            throw new Exception(sprintf(
+                'You must define a $purgeable property in %s to use the Purgeable trait.', get_class($this->parent)
+            ));
+        }
+
+        /*
+         * Remove any purge attributes from the data set
+         */
+        $model = $this->parent;
+        $model->bindEvent('model.saveInternal', function() use ($model) {
+            $model->purgeAttributes();
+        });
+
+    }
+
+    /**
+     * Adds an attribute to the purgeable attributes list
+     * @param  array|string|null  $attributes
+     * @return $this
+     */
+    public function addPurgeable($attributes = null)
+    {
+        $attributes = is_array($attributes) ? $attributes : func_get_args();
+
+        $this->parent->purgeable = array_merge($this->parent->purgeable, $attributes);
+
+        return $this->parent;
+    }
+
+    /**
+     * Removes purged attributes from the dataset, used before saving.
+     * @param $attributes mixed Attribute(s) to purge, if unspecified, $purgable property is used
+     * @return array Current attribute set
+     */
+    public function purgeAttributes($attributesToPurge = null)
+    {
+        if ($attributesToPurge !== null) {
+            $purgeable = is_array($attributesToPurge) ? $attributesToPurge : [$attributesToPurge];
+        }
+        else {
+            $purgeable = $this->getPurgeableAttributes();
+        }
+
+        $dyn_prop_names = $this->parent->getExtensionData()['dynamicProperties'];
```

Comment:
Variables in October generally use camelCase, please change to `dynamicProperties` and just use the `$this->parent->getDynamicProperties()` method